scripts/fetch-microsoft-sdk.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Download a .NET Core SDK published by Microsoft.
+#
+# See --help for more details, including usage
+
+set -euo pipefail
+IFS=$'\n\t'
+
+function usage() {
+    echo ""
+    echo "usage: $0 SDK_VERSION"
+    echo ""
+    echo "Download a particular .NET Core SDK built by Microsoft based on exact version match."
+}
+
+positional_args=()
+while :; do
+    if [ $# -le 0 ]; then
+        break
+    fi
+    lowerI="$(echo "$1" | awk '{print tolower($0)}')"
+    case $lowerI in
+        "-?"|-h|--help)
+            usage
+            exit 0
+            ;;
+        *)
+            positional_args+=("$1")
+            ;;
+    esac
+
+    shift
+done
+
+sdk_version=${positional_args[0]:-}
+if [[ -z ${sdk_version} ]]; then
+    echo "error: missing argument."
+    usage
+    exit 1
+fi
+
+install_script_url="https://dot.net/v1/dotnet-install.sh"
+install_script_dir=$(mktemp -d)
+curl -L "$install_script_url" -o "${install_script_dir}/dotnet-install.sh"
+
+chmod +x "${install_script_dir}/dotnet-install.sh"
+url=$(/tmp/tmp.Jhl1JIGIpU/dotnet-install.sh --dry-run --version "$sdk_version"| grep https | sed -E 's|.*(https://.*tar.gz)$|\1|' | head -1)
```

Comment:
Is this meant to be:

```suggestion
url=$("$install_script_dir/dotnet-install.sh" --dry-run --version "$sdk_version"| grep https | sed -E 's|.*(https://.*tar.gz)$|\1|' | head -1)
```

(Clever approach, I like how it isolates us from funny script behavior.)

In the end, we'll need something special to handle the internal scenario and (if we can) put the Msft SDK in smoke-test prereqs, but managing to implement that without Msft internal access would be magic so IMO getting this impl in for now is perfectly fine. 😄